simulator/mips/mips_register/mips_register.h

```
@@ -41,7 +36,29 @@ class MIPSRegister {
     bool is_zero()       const { return value == MIPS_REG_zero; }
     bool is_mips_hi()    const { return value == MIPS_REG_hi; }
     bool is_mips_lo()    const { return value == MIPS_REG_lo; }
-    size_t to_size_t()   const { return value; }
+
+    static MIPSRegister from_cpu_index( uint8 id) {
+        assert( id < 32u);
+        return MIPSRegister( static_cast<RegNum>( id));
+    }
+    static MIPSRegister from_cp0_index( uint8 id) {
+        /* To be implemented */
+        (void) id; return MIPSRegister::zero;
+    }
+    static MIPSRegister from_gdb_index( uint8 id)
+    {
+        if (id < 32u)
+            return from_cpu_index( id);
+        else switch (id) {
+            case 32: return from_cp0_index( 12); // SR
+            case 33: return MIPSRegister::mips_lo;
+            case 34: return MIPSRegister::mips_hi;
+            case 35: return from_cp0_index( 8);  // Bad
+            case 36: return from_cp0_index( 13); // Cause
+            default: return MIPSRegister::zero;  // CP1 registers etc.
```

Actually. that's the place where we can throw an exception: GDB user want to look at something we cannot show now.